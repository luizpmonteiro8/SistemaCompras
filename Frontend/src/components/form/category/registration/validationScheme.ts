import * as Yup from 'yup';

export const validationScheme = Yup.object().shape({
  name: Yup.string().trim().required('Campo obrigat√≥rio.'),
});
