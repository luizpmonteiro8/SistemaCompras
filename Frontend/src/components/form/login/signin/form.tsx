import { useFormik } from 'formik';
import { Input, InputPassword } from 'components/common/input';
import { useEffect, useState } from 'react';
import { Credential, EmailDTO } from 'app/models/user';
import { validationScheme } from './validationScheme';
import * as Styled from './styles';
import { Modal } from 'react-bootstrap';
import Button from 'react-bootstrap/Button';

export type LoginFormProps = {
  onSubmit: (credential: Credential) => void;
  forgotPassword: (emailDTO: EmailDTO) => void;
};

const formSchema = {
  email: 'teste@teste.com.br',
  password: '12345678',
};

export const LoginForm = ({ onSubmit, forgotPassword }: LoginFormProps) => {
  const [email, setEmail] = useState<EmailDTO>({ email: '' });
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);
  const formik = useFormik<Credential>({
    initialValues: { ...formSchema },
    onSubmit,
    validationSchema: validationScheme,
  });

  useEffect(() => {
    return null;
  }),
    [];

  return (
    <Styled.Form>
      <form className="form-group" onSubmit={formik.handleSubmit}>
        <div>
          <div className="row m-2">
            <div className="col-md-12 ">
              <Input
                id="email"
                name="email"
                onChange={formik.handleChange}
                value={formik.values.email}
                label="Email"
                error={formik.touched.email ? formik.errors.email : ''}
              />
            </div>
            <div className=" col-md-12 ">
              <InputPassword
                id="password"
                label="Senha"
                onChange={formik.handleChange}
                name="password"
                value={formik.values.password}
                error={formik.touched.password ? formik.errors.password : ''}
              />
              <div className="row justify-content-center mt-2">
                <div className="col-md-8  text-center ">
                  <button type="submit" className="btn btn-primary">
                    Entrar
                  </button>
                  <a className="btn btn-danger ms-2" href="/cadastros/usuario">
                    Cadastrar
                  </a>
                  <a className="btn btn-info ms-2" onClick={handleShow}>
                    Recuperar senha
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Recuperar senha</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <Input
            id="email"
            name="email"
            type="email"
            value={email.email}
            onChange={(e) => {
              setEmail({ email: e.target.value });
            }}
            label="Email"
          />
        </Modal.Body>
        <Modal.Footer>
          <Button
            variant="danger"
            onClick={() => {
              forgotPassword(email);
              handleClose();
            }}
          >
            Enviar
          </Button>
          <Button variant="info" onClick={handleClose}>
            Cancelar
          </Button>
        </Modal.Footer>
      </Modal>
    </Styled.Form>
  );
};
