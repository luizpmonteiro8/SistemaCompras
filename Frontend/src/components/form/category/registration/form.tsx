import { useFormik } from 'formik';
import { Input } from 'components/common/input';
import { Category } from 'app/models/category';
import { validationScheme } from './validationScheme';

export type CategoryFormProps = {
  category: Category;
  onSubmit: (category: Category) => void;
};

const formSchema = {
  id: '',
  name: '',
};

export const CategoryForm = ({ category, onSubmit }: CategoryFormProps) => {
  const formik = useFormik<Category>({
    initialValues: { ...formSchema, ...category },
    onSubmit,
    validationSchema: validationScheme,
    enableReinitialize: true,
  });

  return (
    <form className="form-group" onSubmit={formik.handleSubmit}>
      <div>
        <div className="row m-2">
          <div className="col-md-12 ">
            {<Input disabled id="id" name="id" onChange={formik.handleChange} value={formik.values.id} label="Id" />}
          </div>

          <div className="col-md-12 ">
            <Input
              autoFocus={true}
              id="name"
              name="name"
              onChange={formik.handleChange}
              value={formik.values.name}
              error={formik.touched.name && formik.errors.name ? formik.errors.name : ''}
              label="Nome"
            />
          </div>

          <div className="row justify-content-center mt-2">
            <div className="col-md-6  text-center ">
              <button type="submit" className="btn btn-primary">
                Enviar
              </button>
              <a href={'/cadastros/categorias'} className="btn btn-danger ms-2">
                Limpar
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
  );
};
