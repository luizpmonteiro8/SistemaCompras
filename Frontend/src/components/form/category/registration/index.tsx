/* eslint-disable react-hooks/exhaustive-deps */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { useState, useEffect } from 'react';
import { Category } from 'app/models/category';
import { CategoryForm } from './form';
import { useRouter } from 'next/dist/client/router';
import { SaveCategory, UpdateCategory } from 'store/actions/category';
import { connect, ConnectedProps } from 'react-redux';

type Props = PropsFromRedux;

const CategoryRegistration = (props: Props) => {
  const [category, setCategory] = useState<Category>();
  const router = useRouter();
  const { id } = router.query;

  useEffect(() => {
    if (id) {
      props.category.map((item) => {
        if (item.id === parseInt(id.toString())) {
          setCategory(item);
        }
      });
    } else {
      setCategory(null);
    }
  }, [id, props.category]);

  const handleSubmit = async (categoryNew: Category, { resetForm, setValues }) => {
    if (categoryNew.id > 0) {
      if (await props.updateCategory(categoryNew)) {
        resetForm();
        setValues({ id: '', name: '' });
        router.replace('/cadastros/categorias');
      }
    } else {
      if (await props.saveCategory(categoryNew)) {
        resetForm();
      }
    }
  };

  return (
    <div className="card bg-light my-2 mx-auto col-md-8" style={{}}>
      <h4 className="card-header ">Categoria</h4>
      <CategoryForm category={category} onSubmit={handleSubmit} isLoading={props.isLoading} />
      <div className="card-body"></div>
    </div>
  );
};

const mapStateToProps = ({ category }) => {
  return {
    category: category.category as Category[],
    isLoading: category.isLoading,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    saveCategory: (category: Category) => dispatch(SaveCategory(category)),
    updateCategory: (category: Category) => dispatch(UpdateCategory(category)),
  };
};

const connector = connect(mapStateToProps, mapDispatchToProps);
type PropsFromRedux = ConnectedProps<typeof connector>;
export default connector(CategoryRegistration);
