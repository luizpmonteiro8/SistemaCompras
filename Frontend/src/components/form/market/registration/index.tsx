import { useState, useEffect } from 'react';
import { Market } from 'app/models/market';
import { MarketForm } from './form';
import { useRouter } from 'next/dist/client/router';
import { SaveMarket, UpdateMarket } from 'store/actions/market';
import { connect, ConnectedProps } from 'react-redux';

type Props = PropsFromRedux;

const MarketRegistration = (props: Props) => {
  const [market, setMarket] = useState<Market>();
  const router = useRouter();
  const { id } = router.query;

  useEffect(() => {
    if (id) {
      props.market.map((item) => {
        if (item.id === parseInt(id.toString())) {
          setMarket(item);
        }
      });
    }
  }, [id, props.market]);

  const handleSubmit = async (marketItem: Market, { resetForm, setValues }) => {
    if (marketItem.id > 0) {
      if (await props.updateMarket(marketItem)) {
        resetForm({});
        setValues({ id: null, name: '', blocked: false, cnpj: null });
        router.replace('/cadastros/mercados');
      }
    } else {
      if (await props.saveMarket(marketItem)) {
        resetForm({});
        setValues({ id: null, name: '', blocked: false, cnpj: null });
      }
    }
  };

  return (
    <div className="card bg-light my-2 mx-auto col-md-8" style={{}}>
      <h4 className="card-header ">Mercado</h4>
      <MarketForm market={market} onSubmit={handleSubmit} isLoading={props.isLoading} />
      <div className="card-body"></div>
    </div>
  );
};

const mapStateToProps = ({ market }) => {
  return {
    market: market.market as Market[],
    isLoading: market.isLoading,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    saveMarket: (market: Market) => dispatch(SaveMarket(market)),
    updateMarket: (market: Market) => dispatch(UpdateMarket(market)),
  };
};

const connector = connect(mapStateToProps, mapDispatchToProps);
type PropsFromRedux = ConnectedProps<typeof connector>;
export default connector(MarketRegistration);
