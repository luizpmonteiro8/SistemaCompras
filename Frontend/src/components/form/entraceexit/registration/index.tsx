/* eslint-disable react-hooks/exhaustive-deps */
import { Category } from 'app/models/category';
import { Product } from 'app/models/product';
import { EntraceExit } from 'app/models/entraceexit';
import * as Styled from './styles';
import { LoadAllCategory } from 'store/actions/category';
import { LoadAllProduct } from 'store/actions/product';
import { connect, ConnectedProps } from 'react-redux';
import { EntraceExitForm } from './form';
import { useState, useEffect } from 'react';
import { useRouter } from 'next/router';
import { SaveEntraceExit, UpdateEntraceExit } from 'store/actions/entraceexit';

type Props = PropsFromRedux;
const EntraceExitRegistration = (props: Props) => {
  const router = useRouter();
  const { id } = router.query;
  const [entraceExit, setEntraceExit] = useState<EntraceExit>();

  useEffect(() => {
    props.loadAllCategory();
    props.loadAllProduct();

    if (id) {
      props.entraceexit.map((item) => {
        if (item.id === parseInt(id.toString())) {
          setEntraceExit(item);
        }
      });
    }
  }, [id, props.entraceexit]);

  const handleSubmit = async (entraceExitNew: EntraceExit, { resetForm, setValues }) => {
    if (entraceExitNew.id > 0) {
      if (await props.updateEntraceExit(entraceExitNew)) {
        resetForm();
        setValues({ id: '', productId: 0, quantity: 0, type: null });
        router.replace('/cadastros/entradasaida');
      }
    } else {
      if (await props.saveEntraceExit(entraceExitNew)) {
        resetForm();
      }
    }
  };

  const filterProductByCategory = (idCategory) => {
    return props.product.filter((item) => {
      if (item.category.id === idCategory) return item;
    });
  };

  return (
    <Styled.Wrapper>
      <div className="card bg-light my-2 mx-auto col-md-8" style={{}}>
        <h4 className="card-header ">Item da compra</h4>
        <div className="card-body">
          <EntraceExitForm
            category={props.category}
            product={props.product}
            filterProductByCategory={filterProductByCategory}
            entraceExit={entraceExit}
            onSubmit={handleSubmit}
            isLoading={props.isLoading}
          />
        </div>
      </div>
    </Styled.Wrapper>
  );
};

const mapStateToProps = ({ category, product, entraceExit }) => {
  return {
    entraceexit: entraceExit.entraceExit as EntraceExit[],
    category: category.category as Category[],
    product: product.product as Product[],
    isLoading: entraceExit.isLoading,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    loadAllCategory: () => dispatch(LoadAllCategory()),
    loadAllProduct: () => dispatch(LoadAllProduct()),
    saveEntraceExit: (entraceExit) => dispatch(SaveEntraceExit(entraceExit)),
    updateEntraceExit: (entraceExit) => dispatch(UpdateEntraceExit(entraceExit)),
  };
};

const connector = connect(mapStateToProps, mapDispatchToProps);
type PropsFromRedux = ConnectedProps<typeof connector>;
export default connector(EntraceExitRegistration);
