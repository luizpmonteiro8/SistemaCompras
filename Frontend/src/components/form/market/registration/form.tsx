import { useFormik } from 'formik';
import { Input } from 'components/common/input';
import { Market } from 'app/models/market';
import { validationScheme } from './validationScheme';
import { AutoComplete } from 'components/common/autoComplete';

export type MarketFormProps = {
  market: Market;
  onSubmit: (market: Market) => void;
};

const formSchema = {
  id: '',
  name: '',
  blocked: false,
  cnpj: '',
};

export const MarketForm = ({ market, onSubmit }: MarketFormProps) => {
  const formik = useFormik<Market>({
    initialValues: { ...formSchema, ...market },
    onSubmit,
    validationSchema: validationScheme,
    enableReinitialize: true,
  });

  return (
    <form className="form-group" onSubmit={formik.handleSubmit}>
      <div>
        <div className="row m-2">
          <div className="col-md-12 ">
            {<Input disabled id="id" name="id" onChange={formik.handleChange} value={formik.values.id} label="Id" />}
          </div>
          <div className="col-md-12 "></div>
          <div className="col-md-12 ">
            <Input
              autoFocus={true}
              id="name"
              name="name"
              onChange={formik.handleChange}
              value={formik.values.name}
              error={formik.touched.name && formik.errors.name ? formik.errors.name : ''}
              label="Nome"
            />
          </div>
          <div className="col-md-12 ">
            <Input
              autoFocus={true}
              id="cnpj"
              name="cnpj"
              onChange={formik.handleChange}
              value={formik.values.cnpj}
              error={formik.touched.cnpj && formik.errors.cnpj ? formik.errors.cnpj : ''}
              label="CNPJ"
            />
          </div>
          <div className="col-md-12 ">
            <input
              className="form-check-input"
              type="checkbox"
              id="blocked"
              onChange={formik.handleChange}
              defaultChecked={formik.values.blocked ? true : false}
            />
            <label className="form-check-label" htmlFor="blocked">
              Bloqueado
            </label>
          </div>

          <div className="row justify-content-center mt-2">
            <div className="col-md-6  text-center ">
              <button type="submit" className="btn btn-primary">
                Enviar
              </button>
              <a href="/" className="btn btn-danger ms-2">
                Cancelar
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
  );
};
