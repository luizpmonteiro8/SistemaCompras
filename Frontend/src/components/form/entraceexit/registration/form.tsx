import { Input } from 'components/common/input';
import { AutoComplete, dataAutoComplete } from 'components/common/autoComplete';
import { useFormik } from 'formik';
import { EntraceExit } from 'app/models/entraceexit';
import { Product } from 'app/models/product';
import { Category } from 'app/models/category';
import { convertDataAutoComplete } from 'components/common/autoComplete/convertdata';
import { useState } from 'react';
import { validationScheme } from './validationScheme';
import { useRouter } from 'next/dist/client/router';

export type EntraceExitFormProps = {
  category: Category[];
  product: Product[];
  filterProductByCategory: (idCategory: number) => Product[];
  entraceExit: EntraceExit;
  onSubmit: (entraceExit: EntraceExit, { resetForm, setValues }) => void;
};

export const EntraceExitForm = ({
  category,
  product,
  filterProductByCategory,
  entraceExit,
  onSubmit,
}: EntraceExitFormProps) => {
  const formSchema = {
    id: '',
    productId: 0,
    quantity: 0,
    type: null,
    status: true,
  };

  const [categorySelected, setCategorySelected] = useState<dataAutoComplete>();
  const route = useRouter();
  const formik = useFormik<EntraceExit>({
    initialValues: { ...formSchema, ...entraceExit },
    onSubmit,
    validationSchema: validationScheme,
    enableReinitialize: true,
  });

  return (
    <form className="form-group" onSubmit={formik.handleSubmit} noValidate>
      <div>
        <div className="row m-2">
          <div className="col-md-6">
            <Input
              disabled
              id="id"
              name="id"
              onChange={formik.handleChange}
              value={formik.values.id == 0 ? '' : formik.values.id}
              label="Id"
              error={formik.errors.id}
            />
          </div>
        </div>
        <div className="row m-2">
          <div className="col-md-6">
            <AutoComplete
              title="Categoria"
              id="category"
              placeholder="Selecione categoria"
              data={convertDataAutoComplete(category)}
              onChange={(e) => setCategorySelected(e)}
              value={categorySelected}
            />
          </div>

          <div className="col-md-6">
            <AutoComplete
              title="Produto"
              id="type"
              placeholder="Selecione o produto"
              data={convertDataAutoComplete(
                filterProductByCategory(categorySelected?.value).length >= 1
                  ? filterProductByCategory(categorySelected?.value)
                  : product,
              )}
              onChange={(e) => {
                if (e != null) {
                  formik.setFieldValue('productId', e.value);
                } else {
                  formik.setFieldValue('productId', null);
                }
              }}
              idValue={formik.values.productId}
              error={formik.touched.productId && formik.errors.productId ? formik.errors.productId : ''}
            />
          </div>

          <div className="col-md-6 ">
            <Input
              id="quantity"
              name="quantity"
              onChange={(e) => formik.setFieldValue('quantity', e.target.value)}
              label="Quantidade"
              type="number"
              min="0.00"
              step="0.001"
              value={formik.values.quantity}
              error={formik.touched.quantity && formik.errors.quantity ? formik.errors.quantity : ''}
            />
          </div>

          <div className="col-md-6">
            <AutoComplete
              title="Entrada/Saida"
              id="type"
              placeholder="Selecione entrada ou saida"
              data={[
                { value: 0, label: 'Entrada' },
                { value: 1, label: 'Saida' },
              ]}
              onChange={(e: dataAutoComplete) => formik.setFieldValue('type', e.value)}
              idValue={formik.values.type}
              error={formik.touched.type && formik.errors.type ? formik.errors.type : ''}
            />
          </div>

          <div className="row justify-content-center mt-2">
            <div className="col-md-6	text-center ">
              <button type="submit" className="btn btn-primary">
                Enviar
              </button>
              <a
                onClick={() => {
                  route.replace('/cadastros/entradasaida');
                  formik.resetForm();
                  formik.setValues({ id: 0, productId: 0, quantity: 0, type: null, status: true });
                }}
                className="btn btn-danger ms-2"
              >
                Limpar
              </a>
            </div>
          </div>
        </div>
      </div>
    </form>
  );
};
