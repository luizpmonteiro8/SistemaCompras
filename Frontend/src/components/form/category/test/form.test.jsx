/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable @typescript-eslint/no-empty-function */
import { CategoryForm } from 'components/form/category/registration/form';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

jest.mock('next/dist/client/router', () => ({
  useRouter() {
    return {
      replace: jest.fn(),
    };
  },
}));
const mockCategory = { id: 1, name: 'Cereais' };
let useRouter;

describe('<AutoComplete />', () => {
  beforeEach(async () => {
    jest.clearAllMocks();
    useRouter = jest.spyOn(require('next/dist/client/router'), 'useRouter');
    useRouter.mockImplementation(() => ({
      replace: jest.fn(),
    }));
  });

  it('should load mockCategory', async () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<CategoryForm category={mockCategory} onSubmit={onSubmit} isLoading={false} />);

    const nameInput = wrapper.find("input[name='name']").prop('value');
    expect(nameInput).toBe('Cereais');
  });

  it('should change input name to Test', async () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<CategoryForm category={mockCategory} onSubmit={onSubmit} isLoading={false} />);

    const nameInput = wrapper.find("input[name='name']");
    await act(async () => {
      nameInput.simulate('change', {
        persist: () => {},
        target: {
          name: 'name',
          value: 'Test',
        },
      });
    });
    expect(nameInput.html()).toMatch('Test');
  });

  it('should submit form', async () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<CategoryForm category={mockCategory} onSubmit={onSubmit} isLoading={false} />);

    const buttonSubmit = wrapper.find("button[type='submit']");
    await act(async () => {
      buttonSubmit.simulate('submit', {
        persist: () => {},
        target: {
          name: 'name',
          value: 'Test',
        },
      });
    });
    expect(onSubmit).toBeCalledTimes(1);
  });

  it('should call error when submit name empty', async () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<CategoryForm category={null} onSubmit={onSubmit} isLoading={false} />);

    const nameInput = wrapper.find("Input[id='name']");
    const buttonSubmit = wrapper.find("button[type='submit']");
    await act(async () => {
      buttonSubmit.simulate('submit', {
        persist: () => {},
        target: {
          name: 'name',
          value: 'Test',
        },
      });
    });
    expect(onSubmit).toBeCalledTimes(0);
    expect(nameInput.html()).toMatch('Campo obrigatÃ³rio.');
  });

  it('should click button Limpar', async () => {
    const mockRouter = {
      replace: jest.fn(),
    };

    useRouter.mockReturnValue(mockRouter);

    const onSubmit = jest.fn();
    const wrapper = mount(<CategoryForm category={null} onSubmit={onSubmit} isLoading={false} />);

    const buttonClean = wrapper.find('a');
    await act(async () => {
      buttonClean.simulate('click');
    });
    expect(mockRouter.replace).toHaveBeenCalledWith('/cadastros/categorias');
  });
});
