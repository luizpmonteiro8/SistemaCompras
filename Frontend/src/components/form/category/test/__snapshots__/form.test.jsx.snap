// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutoComplete /> should load mockCategory 1`] = `ReactWrapper {}`;
